azure-pipelines.yml

```
@@ -27,36 +28,38 @@ stages:
             displayName: 'Setup TMP Directory'
           - checkout: self
             clean: true
-            submodules: false
+            submodules: true
           - script: |
-              if exist "$(buildOutputLocation)" (
-                rmdir /S /Q "$(buildOutputLocation)"
-              )
               call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\Common7\Tools\VsDevCmd.bat" ^
-              -host_arch=amd64 -arch=amd64 -no_logo
-              cmake -G Ninja -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release ^
-              -S $(Build.SourcesDirectory)\tools -B $(buildOutputLocation)
-              cmake --build $(buildOutputLocation)
-            displayName: 'Build Support Tools'
-            env: { TMP: $(tmpDir), TEMP: $(tmpDir) }
-          - task: BatchScript@1
-            displayName: 'Enforce clang-format'
-            timeoutInMinutes: 60
+                -host_arch=amd64 -arch=amd64 -no_logo
+              cmake -G Ninja -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release -DERROR_ON_CLANG_FORMAT_NOT_FOUND=1 ^
+                -S $(Build.SourcesDirectory) -B $(tmpDir)\format-build
+              cmake --build $(tmpDir)\format-build --target format
+            displayName: 'clang-format'
+            timeoutInMinutes: 5
             condition: succeededOrFailed()
-            inputs:
-              filename: 'azure-devops/enforce-clang-format.cmd'
-              failOnStandardError: true
-              arguments: '$(buildOutputLocation)/parallelize/parallelize.exe'
             env: { TMP: $(tmpDir), TEMP: $(tmpDir) }
           - task: BatchScript@1
-            displayName: 'Validate Files'
-            timeoutInMinutes: 2
-            condition: succeededOrFailed()
             inputs:
               filename: 'azure-devops/validate-files.cmd'
               failOnStandardError: true
-              arguments: '$(buildOutputLocation)/validate/validate.exe'
+              arguments: '$(Build.SourcesDirectory)'
+            displayName: 'Validate Files'
+            timeoutInMinutes: 5
+            condition: succeededOrFailed()
             env: { TMP: $(tmpDir), TEMP: $(tmpDir) }
+          - task: Powershell@2
+            displayName: 'Create Diff'
+            inputs:
+              filePath: azure-devops/Create-PRDiff.ps1
+              arguments: '-DiffFile $(DiffFile)'
+              pwsh: true
+          - task: PublishBuildArtifacts@1
+            condition: failed()
+            displayName: 'Publish Format and Messages File Diff'
+            inputs:
+              PathtoPublish: '$(DiffFile)'

```

I believe that this is case-insensitive, but it should be consistently camel-cased:
```suggestion
              PathToPublish: '$(DiffFile)'
```